[WIP] Rebase

aeroxy opened the pull request.

aeroxy committed:
agent: Implement 4-D methodology in system prompt

- Add primary directive for 4-D methodology (Deconstruct, Diagnose,
  Develop, Deliver)

aeroxy committed:
ai: Fix Open AI Compatible provider stream deserialization error

Make Usage struct fields optional and add flexible JSON deserialization
to ResponseStreamEvent to handle Open AI Compatible provider's custom fields and
empty usage objects.

Fixes deserialization error: "data did not match any variant of untagged
enum ResponseStreamResult"

aeroxy committed:
git: Fix test timeouts by ensuring job queue always processes

The spawn_local_git_worker would exit early if initialization failed
(via ? operators), leaving jobs unprocessed and causing barrier()
calls to hang forever.

Restructured to wrap initialization in a Result and always run the
job processing loop, even when backend initialization fails. Jobs
only execute with a valid state; otherwise they're dropped and
receivers detect the error.

Fixes test_git_repository_status and test_file_status timeouts.

aeroxy committed:
git: Fix borrow after move error in git worker

Clone dot_git_abs_path before it's moved into the async block
so it can be used in error logging.

aeroxy committed:
refactor: initialize usage token counts as `Some(0)`

aeroxy committed:
feat: Implement OpenAI compatible event mapper to parse thinking tags

aeroxy committed:
chore: Refine OpenAI event processing

aeroxy committed:
ai: OpenAI provider supports reasoning content

aeroxy committed:
agent_ui: Add regeneration button to text and agent thread titles

aeroxy committed:
chore: Reset git_store.rs to main

aeroxy committed:
Update .gitignore

aeroxy committed:
language_models: Resolve provider ID conflicts for openai_compatible

aeroxy committed:
deepseek: handle missing index field in tool call responses

- Add #[serde(default)] to ToolCallChunk.index field - Prevents
deserialization errors when DeepSeek API omits index - Maintains
backward compatibility with existing code - Uses default value of 0 when
index is missing

Fixes: "missing field 'index' at line 1 column 505" error

osiewicz commented:
Please do not modify our `ci.yml` in order to run the bundling. This is a pretty malicious behavior. It is not the first time it happened either (see [this PR](https://github.com/zed-industries/zed/pull/42531/files)).

aeroxy commented:
Is there a method to run the bundling officially? I don't have enough disk
space locally to run the bundling locally so I need some way to verify the
build.

On Wed, Dec 10, 2025, 11:40 PM Piotr Osiewicz ***@***.***>
wrote:

> *osiewicz* left a comment (zed-industries/zed#44545)
> <https://github.com/zed-industries/zed/pull/44545#issuecomment-3637700243>
>
> Please do not modify our ci.yml in order to run the bundling. This is a
> pretty malicious behavior. It is not the first time it happened either (see this
> PR <https://github.com/zed-industries/zed/pull/42531/files>).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zed-industries/zed/pull/44545#issuecomment-3637700243>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAVCEW2LH34D2M5JDO24T4T4BA5IRAVCNFSM6AAAAACOTZCMGKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMMZXG4YDAMRUGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

aeroxy committed:
language_models: Fix build error

aeroxy committed:
language_models: Fix test error

aeroxy committed:
chore: Format

aeroxy committed:
open_ai: Support DeepSeek V3.2 in openAI compatiable provider

aeroxy committed:
chore: Update the usage token type

aeroxy committed:
open_ai: Parse a response line with more lenient error handling for different providers

aeroxy committed:
markdown_preview: Add Copy All action

Add action to copy all markdown preview content to clipboard as plain
text. Accessible via command palette.

Also fixes fsevent test variable shadowing issue.

aeroxy committed:
chore: Update optional u64 to u64

aeroxy committed:
markdown_preview: Fix the merge errors

aeroxy committed:
chore: Remove unused actions

aeroxy committed:
open_ai: Fix tool calls parsing for malformed data from different
providers

aeroxy committed:
open_id: Enhance tool calls robustness for DeepSeek V3.2

aeroxy committed:
chore: Update token type to some u64

aeroxy committed:
open_ai: Allow DeepSeek V3.2 tool calls

aeroxy committed:
open_ai: Remove unused ToolCallsHelper and allow duplicated clones

aeroxy committed:
protoc: Better protobuf detection

aeroxy committed:
chore: Format

aeroxy committed:
open_ai: Support tool calls streaming

aeroxy committed:
build: Better protobuf detection

aeroxy committed:
agent_ui: For short selections (like keywords) show the text instead of the line number

aeroxy committed:
build: Better protobuf detection

aeroxy committed:
Revert "Better protobuf detection"

aeroxy committed:
open_ai: Allow DeepSeek V3.2 streaming tool calls

aeroxy committed:
open_ai: Fix streaming tool calls handling

aeroxy committed:
test
